closed/make/Javadoc.gmk

```diff
@@ -0,0 +1,490 @@
+#
+# The OpenJ9 classes are built from the j9jcl_sources and closed/src
+# directories.
+# The packages are beneath directories called /share/classes (or /unix/classes,
+# /windows/classes etc.) which do not appear in the javadoc tree structure.
+# The commands below replace the footer in the javadoc html pages for the
+# following files:
+# 1. The openj9 java source files which override the openjdk versions
+# 2. The openj9.xxxx modules, including the javadoc generated files such as
+#    enum class pages, module-summary.html, package-summary.html etc.)
+# A temporary script file is used to avoid the xargs command line length limit
+# of 255 chars.
+
+OPENJ9_SUBST_BOTTOM = $(subst &,\&,$(OPENJ9_JAVADOC_BOTTOM))
+
+#############################################################################
+# sed command to replace the footer in the jdk management javadoc files.
+MGMT_BOTTOM_REGEX = $(subst .,\.,$(MGMT_BOTTOM))
+OPENJ9_MGMT_SUBST_COMMAND = $(SED) -e '\''s|$(MGMT_BOTTOM_REGEX)|$(OPENJ9_SUBST_BOTTOM)|g'\''
+
+#############################################################################
+# sed command to replace the footer in the core api javadoc files.
+# Openjdk Core API doc is generated with the -Xdocrootparent javadoc option,
+# so apply the same substition to the page footer before matching.
+COREAPI_BOTTOM_REGEX = $(subst {@docroot}/,$(subst .,\.,$(DOCS_BASE_URL)),$(subst .,\.,$(COREAPI_BOTTOM)))
+OPENJ9_COREAPI_SUBST_COMMAND = $(SED) -e '\''s|$(COREAPI_BOTTOM_REGEX)|$(OPENJ9_SUBST_BOTTOM)|g'\''
+
+################################################################################
+OPENJ9_JAVADOC_MARKER := $(TEMPDIR)/_javadoc_openj9.marker
+OPENJ9_JAVADOC_SCRIPT := $(TEMPDIR)/_javadoc_openj9_footer_replace.sh
+OPENJ9_JAVADOC_TEMP  := $(TEMPDIR)/_javadoc_temp.html
+
+################################################################################
+# Use a marker file as target with dependencies which ensure footers are not
+# updated until the pages have been updated by the openjdk javadoc targets. 
+
+OPENJ9_JAVADOC_SCRIPT_BODY = \
+    for path in "$$@" ; do \
+      if [ -f "$(COREAPI_DOCSDIR)/$$path" ] ; then \
+        $(CP) "$(COREAPI_DOCSDIR)/$$path" "$(OPENJ9_JAVADOC_TEMP)" ; \
+        $(OPENJ9_COREAPI_SUBST_COMMAND) < "$(OPENJ9_JAVADOC_TEMP)" > "$(DOCS_OUTPUTDIR)/$$path" ; \
+        $(RM) "$(OPENJ9_JAVADOC_TEMP)" ; \
+      fi ; \
+      if [ -f "$(JRE_API_DOCSDIR)/management/extension/$$path" ] ; then \
+        $(CP) "$(JRE_API_DOCSDIR)/management/extension/$$path" "$(OPENJ9_JAVADOC_TEMP)" ; \
+        $(OPENJ9_MGMT_SUBST_COMMAND) < "$(OPENJ9_JAVADOC_TEMP)" > "$(JRE_API_DOCSDIR)/management/extension/$$path" ; \
+        $(RM) "$(OPENJ9_JAVADOC_TEMP)" ; \
+      fi ; \
+    done \
+    #
+
+$(OPENJ9_JAVADOC_MARKER) : $(COREAPI_DOCSDIR)/index.html openj9-mgmtdocs
+	$(ECHO) '$(OPENJ9_JAVADOC_SCRIPT_BODY)' >$(OPENJ9_JAVADOC_SCRIPT)
+	$(CD) $(SUPPORT_OUTPUTDIR)/j9jcl_sources ; \
```

Comment:
The directory should be `$(JDK_OUTPUTDIR)/j9jcl_sources`.

Reply:
Apologies, I had neglected to check in the final version of the code.  Those directories are now fixed.